samples/nrf_desktop/src/modules/usb_state.c

```diff
@@ -91,36 +91,55 @@ static void send_mouse_report(const struct hid_mouse_event *event)
 		return;
 	}
 
-	s16_t wheel = max(min(event->wheel, REPORT_MOUSE_WHEEL_MAX),
-			  REPORT_MOUSE_WHEEL_MIN);
-	s16_t x = max(min(event->dx, REPORT_MOUSE_XY_MAX),
-		      REPORT_MOUSE_XY_MIN);
-	s16_t y = max(min(event->dy, REPORT_MOUSE_XY_MAX),
-		      REPORT_MOUSE_XY_MIN);
-
-	/* Convert to little-endian. */
-	u8_t x_buff[2];
-	u8_t y_buff[2];
-
-	sys_put_le16(x, x_buff);
-	sys_put_le16(y, y_buff);
-
-	/* Encode report. */
-	u8_t buffer[REPORT_SIZE_MOUSE + sizeof(u8_t)];
-
-	static_assert(sizeof(buffer) == 6, "Invalid report size");
-
-	buffer[0] = REPORT_ID_MOUSE;
-	buffer[1] = event->button_bm;
-	buffer[2] = wheel;
-	buffer[3] = x_buff[0];
-	buffer[4] = (y_buff[0] << 4) | (x_buff[1] & 0x0f);
-	buffer[5] = (y_buff[1] << 4) | (y_buff[0] >> 4);
-
-	int err = hid_int_ep_write(buffer, sizeof(buffer), NULL);
-	if (err) {
-		LOG_ERR("Cannot send report (%d)", err);
-		mouse_report_sent(true);
+	if (hid_protocol == HID_PROTOCOL_REPORT) {
+		s16_t wheel = max(min(event->wheel, REPORT_MOUSE_WHEEL_MAX),
+				REPORT_MOUSE_WHEEL_MIN);
+		s16_t x = max(min(event->dx, REPORT_MOUSE_XY_MAX),
+				REPORT_MOUSE_XY_MIN);
+		s16_t y = max(min(event->dy, REPORT_MOUSE_XY_MAX),
+				REPORT_MOUSE_XY_MIN);
+		/* Convert to little-endian. */
+		u8_t x_buff[2];
+		u8_t y_buff[2];
+
+		sys_put_le16(x, x_buff);
+		sys_put_le16(y, y_buff);
+
+		/* Encode report. */
+		u8_t buffer[REPORT_SIZE_MOUSE + sizeof(u8_t)];
+
+		static_assert(sizeof(buffer) == 6, "Invalid report size");
+
+		buffer[0] = REPORT_ID_MOUSE;
+		buffer[1] = event->button_bm;
+		buffer[2] = wheel;
+		buffer[3] = x_buff[0];
+		buffer[4] = (y_buff[0] << 4) | (x_buff[1] & 0x0f);
+		buffer[5] = (y_buff[1] << 4) | (y_buff[0] >> 4);
+
+		int err = hid_int_ep_write(buffer, sizeof(buffer), NULL);
```

Comment:
`hid_int_ep_write` is a common part - can be placed after both cases

when you declare the buffer you can do
`u8_t buffer[(boot)?(x):(y)];`

Reply:
done